core/src/main/java/hudson/model/User.java

```diff
@@ -370,21 +372,37 @@ public synchronized void addProperty(@Nonnull UserProperty p) throws IOException
     }
 
     /**
-     * This method checks with {@link SecurityRealm} if the user is a valid user that can login to the security realm.
-     * If {@link SecurityRealm} is a kind that does not support querying information about other users, this will
-     * use {@link LastGrantedAuthoritiesProperty} to pick up the granted authorities as of the last time the user has
-     * logged in.
-     *
+     * This method checks with ${@link UserDetailsCache} and {@link SecurityRealm} if the user is a valid user that can
+     * login to the security realm. 
+     * 
      * @return userDetails for the user, in case he's not found but seems legitimate, we provide a userDetails with minimum access
      * @throws UsernameNotFoundException If this user is not a valid user in the backend {@link SecurityRealm}.
+     * @see #getUserDetailsForImpersonation(boolean)
      */
     public @Nonnull UserDetails getUserDetailsForImpersonation() throws UsernameNotFoundException {
-        ImpersonatingUserDetailsService userDetailsService = new ImpersonatingUserDetailsService(
-                Jenkins.get().getSecurityRealm().getSecurityComponents().userDetails
-        );
+        return getUserDetailsForImpersonation(true);
+    }
 
+    /**
+     * This method checks with ${@link UserDetailsCache} and {@link SecurityRealm} if the user is a valid user that 
```

Comment:
```suggestion
     * This method checks with {@link UserDetailsCache} and {@link SecurityRealm} if the user is a valid user that 
```

nit: correct JavaDoc